src/sdk/index.js

```diff
@@ -4,23 +4,42 @@ import {merge} from 'lodash'
 import api from './api'
 import {DEFAULT_HEADERS} from './headers'
 
+function configGlobalSpec (globalSpec) {
+
+  let headers
+  if (typeof globalSpec.headers === 'function') {
+    headers = {...globalSpec.headers()}
+  }
+  else {
+    headers = globalSpec.headers
+  }
+
+  return {
+    ...globalSpec,
+    ...{headers}
+  }
+}
+
 export default function sdk (globalSpec = {}) {
 
   let defaultSpec = {
     origin: '',
     headers: DEFAULT_HEADERS
   }
-  let options = merge({}, defaultSpec, globalSpec)
+
+  let options = merge({}, defaultSpec, configGlobalSpec(globalSpec))
+
   let stream = Observable.create()
 
   function apiFactory (instanceSpec = {}) {
 
-  	return api(merge({}, options, instanceSpec, {
-  		getStream: () => stream
-  	}))
+    options = merge({}, defaultSpec, configGlobalSpec(globalSpec))
```

Comment:
sound it use a different variable, then? Or line 30 could be defined as a function that's called in apiFactory as well as in apiFactory.createStream.